pandas/tests/copy_view/test_methods.py

```diff
@@ -526,14 +526,15 @@ def test_shift_rows_freq(using_copy_on_write):
     tm.assert_frame_equal(df2, df_orig)
 
 
-def test_shift_columns(using_copy_on_write):
+def test_shift_columns(using_copy_on_write, warn_copy_on_write):
     df = DataFrame(
         [[1, 2], [3, 4], [5, 6]], columns=date_range("2020-01-01", "2020-01-02")
     )
     df2 = df.shift(periods=1, axis=1)
 
     assert np.shares_memory(get_array(df2, "2020-01-02"), get_array(df, "2020-01-01"))
-    df.iloc[0, 0] = 0
+    with tm.assert_cow_warning(warn_copy_on_write):
```

Comment:
Is shift a view in non-cow?

Reply:
This is for axis=1 (shifting rows would never be possible as a view), and apparently yes:

```
In [20]:     df = DataFrame(
    ...:         [[1, 2], [3, 4], [5, 6]], columns=date_range("2020-01-01", "2020-01-02")
    ...:     )
    ...:     df2 = df.shift(periods=1, axis=1)

In [21]: df
Out[21]: 
   2020-01-01  2020-01-02
0           1           2
1           3           4
2           5           6

In [22]: df2
Out[22]: 
   2020-01-01  2020-01-02
0         NaN           1
1         NaN           3
2         NaN           5

In [23]: df2.iloc[0, 1] = 100

In [24]: df
Out[24]: 
   2020-01-01  2020-01-02
0         100           2
1           3           4
2           5           6
```

Reply:
Unexpected, thx